src/api/artifact/model.go

```diff
@@ -43,9 +49,15 @@ type ResourceLink struct {
 // Option is used to specify the properties returned when listing/getting artifacts
 type Option struct {
```

Comment:
For the arguments in the calling method, it can be

```go
func (r *Receiver) Call(arg1 interface{}, options ...Option) error {
    ops := &Options{}
    for _, op := range options {
      op(ops)
    }
}
```

Reply:
```go
 UpperCall(
  param1, 
  WithTag("tag"), 
  WithSignature(signature),
)
```

Reply:
Not very necessary for now.

This pattern would make sense when options has a long list of different type if options, like creating a file or container.

I would suggest we only do such optimization when we need to...

Reply:
Some references:

https://commandcenter.blogspot.com/2014/01/self-referential-functions-and-design.html
https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis